### PR TITLE
Add support for local registry use to the local-dev tooling

### PR DESCRIPTION
**Pull request by malancas**
#### Summary

Adds support for using the local Kind registry. This pull request also adds a new `k8s-version` flag. If supplied, this option determines which Kind image is used when creating the Kind cluster.

cc @hectorj2f @vaikas

### CODE REVIEW

File: hack/local-dev.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# Copyright 2022 The Sigstore Authors
```
Comment (hectorj2f) [inline hack/local-dev.sh:3]: Copyright 2023

File: hack/local-dev.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+# The script accepts the following optional arguments:
+# 1. cluster-name - default value is 'policy-controller-demo'
+# 2. k8s-version - default value is 'v1.24.x'
+# 3. registry-url - if no value is provided, the local Kind registry is used
+#
+
+set -o errexit
+
+CLUSTER_NAME="policy-controller-demo"
+K8S_VERSION="v1.24.x"
```
Comment (hectorj2f) [inline hack/local-dev.sh:26]: Could we default to 1.26.X ?

File: hack/local-dev.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+fi
+
+export GIT_HASH=$(git rev-parse HEAD)
+export GIT_VERSION=$(git describe --tags --always --dirty)
+
+CONFIG_FILES=$(find config -name "*.yaml" ! -name 'kustomization.yaml' | sort)
+
+for i in ${CONFIG_FILES[@]}
+do
+  ko apply -f $i
```
Comment (hectorj2f) [inline hack/local-dev.sh:129]: Why don't you run `make ko-apply` instead ?

**Review by cpanato — COMMENTED** [review]
Personally i think we should stay using go instead of bash

**Comment by hectorj2f** [comment]
I am okay with any approach as long as it is easy to mantain in a longer term.i don't have strong preferences here.

**Comment by malancas** [comment]
Thanks for the feedback, I was able to implement more with the Go CLI. I'll push an update soon that includes it.

**Comment by cpanato** [comment]
cool!

File: cmd/local-dev/setup.go [hunk 4]
```diff
@@ -17,21 +17,46 @@ package main
...
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."%s:%d"]
+    endpoint = ["http://%s:%d"]
+`
+
 func addSetupFlags(cmd *cobra.Command) {
 	cmd.Flags().String("cluster-name", "policy-controller-demo", "name of the dev policy controller cluster")
-	cmd.Flags().String("ko-docker-repo", "", "name of the Ko Docker repository to use")
-	cmd.MarkFlagRequired("ko-docker-repo") //nolint:errcheck
+	cmd.Flags().String("registry-url", "registry.local", "URL of the Ko Docker registry to use. If no registry is provided, the local Kind registry will be used")
+	cmd.Flags().String("k8s-version", "v1.26.x", "name of the Ko Docker repository to use")
```
Comment (hectorj2f) [inline cmd/local-dev/setup.go:59]: If we use `v1.x.x` format, we should explain the other valid values. It isn't clear to me which is a valid k8s version. Perhaps we should use regexp to valid the chosen version.

File: cmd/local-dev/setup.go [hunk 5]
```diff
@@ -121,6 +171,68 @@ func setup() {
...
+	}
+
+	fmt.Println("Connecting network between kind with local registry ...")
+
+	return cli.NetworkConnect(context.Background(), "kind", localRegistryName, nil)
+}
+
+func getKindImage(k8sVersion string) string {
+	switch k8sVersion {
+	case "v1.23.x":
```
[the remark this answers is not in the document]
Comment (malancas) [reply cmd/local-dev/setup.go:213]: Sure, I'll look into handling this with regexp.

**Review by hectorj2f — APPROVED** [review]

**Comment by malancas** [comment]
thanks for the review @hectorj2f, would you mind merging the PR? I don't have access to merge it

**Review by cpanato — APPROVED** [review]

**Pull request merged**